src/dataframe/dataframe.jl

```diff
@@ -264,28 +264,24 @@ function Base.getindex(df::DataFrame, row_ind::Integer, col_ind::ColumnIndex)
     return columns(df)[selected_column][row_ind]
 end
 
-# df[SingleRowIndex, MultiColumnIndex] => DataFrame
-function Base.getindex(df::DataFrame, row_ind::Bool, col_inds::AbstractVector)
-    throw(ArgumentError("invalid row index: $row_ind of type Bool"))
-end
-
 # df[SingleRowIndex, MultiColumnIndex] => DataFrame (will be NamedTuple)
 function Base.getindex(df::DataFrame, row_ind::Integer, col_inds::AbstractVector)
-    Base.depwarn("Selecting a single row from a `DataFrame` will return a `NamedTuple` in future.", :getindex)
+    if row_ind isa Bool
+        throw(ArgumentError("invalid row index: $row_ind of type Bool"))
+    end
+    Base.depwarn("Selecting a single row from a `DataFrame` will return a `NamedTuple` in the future.", :getindex)
```

Comment:
There are still several messages where it would be useful to give the syntax which allows avoiding the warning and getting the same result as currently in the future.